src/main/java/com/puppycrawl/tools/checkstyle/XmlLoader.java

```diff
@@ -90,8 +83,13 @@ public void parseInputSource(InputSource inputSource)
     public InputSource resolveEntity(String publicId, String systemId)
             throws SAXException, IOException {
         final InputSource inputSource;
-        final String dtdResourceName =
-            publicIdToResourceNameMap.get(publicId);
+        final String dtdResourceName;
+        if (publicId == null) {
+            dtdResourceName = null;
```

Comment:
`HashMap` accepts `null` as the key, but the map returned by `Map.copyOf` doesn't.